projects/step-frontend/src/lib/modules/timeseries/components/dashboard/dashboard-state-engine.ts

```diff
@@ -40,15 +41,38 @@ export class DashboardStateEngine {
     if ((state.refreshInProgress || state.context.getChartsLockedState()) && !force) {
       return;
     }
-    let newRange: TimeRange;
-    if (state.timeRangeSelection.type === 'FULL' && state.context.defaultFullTimeRange?.from) {
-      const defaultFullTimeRange = state.context.defaultFullTimeRange;
-      newRange = { from: defaultFullTimeRange.from!, to: defaultFullTimeRange.to || new Date().getTime() };
-    } else {
-      newRange = TimeSeriesUtils.convertSelectionToTimeRange(state.timeRangeSelection);
+    const timeRangeSettings = state.context.getTimeRangeSettings();
+    switch (timeRangeSettings.type) {
+      case 'FULL':
+        if (timeRangeSettings.defaultFullRange?.from && !timeRangeSettings.defaultFullRange?.to) {
+          // to has to be set to now
+          timeRangeSettings.fullRange.to = new Date().getTime() - 5000;
+        }
+        break;
+      case 'ABSOLUTE':
+        break;
+      case 'RELATIVE':
+        const now = new Date().getTime() - 5000;
+        let isFullTimeSelection =
+          timeRangeSettings.selectedRange.from === timeRangeSettings.fullRange.from &&
+          timeRangeSettings.selectedRange.to === timeRangeSettings.fullRange.to;
+        const fullRange = { from: now - timeRangeSettings.relativeSelection!.timeInMs, to: now };
+        timeRangeSettings.fullRange = fullRange;
+        if (isFullTimeSelection) {
+          timeRangeSettings.selectedRange = { ...timeRangeSettings.fullRange };
+        } else {
+          // we have a custom selection. has to be cropped if needed
+          const currentSelection = timeRangeSettings.selectedRange;
+          const newFrom = Math.max(fullRange.from!, currentSelection.from!);
+          const newTo = Math.min(fullRange.to!, currentSelection.to!);
+          if (newTo - newFrom < 3) {
+            timeRangeSettings.selectedRange = { ...timeRangeSettings.fullRange }; // zoom reset when the interval is very small
```

Comment:
```suggestion
            // zoom reset when the interval is very small
            timeRangeSettings.selectedRange = { ...timeRangeSettings.fullRange }; 
```

Reply:
The 3 looks quite random to me, why must the zoom be reset for 2 but not for lets say 5?

Reply:
@neogucky I added a variable and also added a comment. I tried to explain but I don't remember exactly the context, but I've used that for a reason